bus/client.go

```diff
@@ -146,9 +149,51 @@ func (c *Client) SendSiacoins(ctx context.Context, scos []types.SiacoinOutput) (
 	return c.BroadcastTransaction(ctx, append(parents, txn))
 }
 
+// WalletTransactionsOption is an option for the WalletTransactions method.
+type WalletTransactionsOption func(url.Values)
+
+func WalletTransactionsWithBefore(since time.Time) WalletTransactionsOption {
+	return func(q url.Values) {
+		q.Set("before", since.Format(time.RFC3339))
+	}
+}
+
+func WalletTransactionsWithSince(since time.Time) WalletTransactionsOption {
+	return func(q url.Values) {
+		q.Set("since", since.Format(time.RFC3339))
+	}
+}
+
+func WalletTransactionsWithLimit(limit int) WalletTransactionsOption {
+	return func(q url.Values) {
+		q.Set("limit", fmt.Sprint(limit))
+	}
+}
+
+func WalletTransactionsWithOffset(offset int) WalletTransactionsOption {
+	return func(q url.Values) {
+		q.Set("offset", fmt.Sprint(offset))
+	}
+}
+
 // WalletTransactions returns all transactions relevant to the wallet.
-func (c *Client) WalletTransactions(ctx context.Context, since time.Time, max int) (resp []wallet.Transaction, err error) {
-	err = c.c.WithContext(ctx).GET(fmt.Sprintf("/wallet/transactions?since=%s&max=%d", api.ParamTime(since), max), &resp)
+func (c *Client) WalletTransactions(ctx context.Context, opts ...WalletTransactionsOption) (resp []wallet.Transaction, err error) {
+	c.c.Custom("GET", "/wallet/transactions", nil, &resp)
```

Comment:
Why did you go `Custom` ? I believe this should work:
```go
values := url.Values{}
for _, opt := range opts {
    opt(values)
}
err = c.c.WithContext(ctx).GET("/wallet/transactions?"+values.Encode(), &resp)
return
```

Reply:
adding this also gets rid of `c.do` which wasn't reused

Reply:
It does work. The plan was to isolate everything in a `c.do` for now and then try and get Luke to add it to jape to avoid reusing the same code over and over.

Reply:
I don't understand why that is better than the snippet I posted above? I don't see the upside in adding `c.do` to `jape` because we'd have to manually construct the url and set the query string on it using `u.RawQuery = values.Encode()`